todolist/client/src/main/scala/handlers/PingPongClientHandler.scala

```diff
@@ -22,16 +22,21 @@ import cats.effect.{Resource, Sync}
 import examples.todolist.client.clients.PingPongClient
 import examples.todolist.protocol.Protocols.PingPongService
 import higherkindness.mu.rpc.protocol.Empty
-import org.log4s._
+import io.chrisdavenport.log4cats.Logger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
 
 class PingPongClientHandler[F[_]: Sync](client: Resource[F, PingPongService[F]])
     extends PingPongClient[F] {
 
-  val logger: Logger = getLogger
+  implicit def unsafeLogger[L[_]: Sync] = Slf4jLogger.getLogger[F]
 
   override def ping(): F[Unit] =
     client
       .use(_.ping(Empty))
-      .map(p => logger.info(s"Pong received with timestamp: ${p.time}"))
-
+      // compiler complains about a discarded non-unit value below, which makes me think:
+      // since this method isn't pure is it even worth
+      // using log4cats?  The only benefit is moving from two logging libs to one
+      // for this module, which seems like a benefit to me
+      // I also don't understand why Logger[F].info doesn't return unit...
```

Comment:
Open question for whomever wants to take it; my lack of experience with log4cats is showing; does anyone have any advice on modifying the code to satisfy the compiler?

Reply:
The logging methods return `F[Unit]` because log4cats treats logging as a side-effect.

In this case you want `client.use(...).flatMap(p => ...)` instead of `map`.

Reply:
Thank you!